astor/code_gen.py

```diff
@@ -380,6 +380,28 @@ def visit_If(self, node):
                 self.else_body(else_)
                 break
 
+    # introduced in Python 3.10
+    def visit_Match(self, node):
+        # set_precedence(node, node.subject)
+        self.statement(node, 'match ', node.subject, ':')
+        self.body(node.cases)
+
+    def visit_match_case(self, node):
+        # set_precedence(node, node.pattern, node.guard)
+        if node.guard is not None:
+            self.statement(node, 'case ', node.pattern, ' if ', node.guard, ':')
+        else:
+            self.statement(node, 'case ', node.pattern, ':')
+        self.body(node.body)
+
+    def visit_MatchAs(self, node):
+        self.write(node.pattern, ' as ', node.name)
```

Comment:
Empty `MatchAs` should become `_`
https://docs.python.org/3/library/ast.html#ast.MatchAs